tests/useImplementingTypes/spec.ts

```diff
@@ -0,0 +1,43 @@
+import { plugin } from '../../src';
+import testSchema from './schema';
+
+it('should support useImplementingTypes', async () => {
+    const result = await plugin(testSchema, [], { prefix: 'mock', useImplementingTypes: true });
+
+    expect(result).toBeDefined();
+    // Boolean
+    expect(result).toContain(
+        "config: overrides && overrides.hasOwnProperty('config') ? overrides.config! : mockTestAConfig() || mockTestTwoAConfig(),",
+    );
+
+    expect(result).toMatchSnapshot();
+});
+
+it(`shouldn't support useImplementingTypes`, async () => {
+    const result = await plugin(testSchema, [], { prefix: 'mock' });
+
+    expect(result).toBeDefined();
+    // Boolean
+    expect(result).toContain(
+        "config: overrides && overrides.hasOwnProperty('config') ? overrides.config! : mockAConfig(),",
+    );
+
+    expect(result).toMatchSnapshot();
+});
+
+it(`shouldn't support useImplementingTypes with fieldGeneration prop`, async () => {
+    const result = await plugin(testSchema, [], {
+        prefix: 'mock',
+        useImplementingTypes: true,
+        fieldGeneration: {
+            A: { config: 'email' },
+        },
```

Comment:
What if we want to use `useImplementingTypes` (for `config`) and want to use `fieldGeneration` for another field.
Like:
```
        useImplementingTypes: true,
        fieldGeneration: {
            A: { str: 'email' },
        },
```


I think that could make sense, but your code will disable implementing types if `fieldGeneration` is defined.
You should instead disable `useImplementingTypes` only for properties that are overridden by `fieldGeneration

Reply:
done.